Fixed & Tested: Udp-Throughput and TCP-Throughput finalized

monfron opened the pull request:
- The following is ADDED:
     - client netcode for udp-throughput msmt
     - server netcode for udp-throughput msmt
- The following is FIXED:
     - at the server side the select + default (incl. read operation) statement 
     is removed for Tcp and Udp msmt
     - as a result: the default case does not block "infinitly" => the close cmd can be received
- The following is TESTED:
     - Tcp standalone
     - UDP standalone
     - Tcp multiple operations
     - Udp multiple operation
     - Tcp and udp mixed
- WHAT DOES NOT WORK:
     - Still I didnt changed the makefile 
     - Still the relative path is not included 
(c.f. https://stackoverflow.com/questions/38517593/relative-imports-in-go)
     - to BUILD THE PROJECT perform: make install

monfron committed:
Added: udp-throughput/client.go

- similar to the tcp msmt module the client now has the netcode
for udp-throughput

Signed-off-by: monfron <simon.fronhoefer@gmail.com>

monfron committed:
Added: start udp-msmt at mapago-client.go, func prototyp for managing udp conn

Signed-off-by: monfron <simon.fronhoefer@gmail.com>

monfron committed:
Added: manage Udp msmt by mapago-client.go, func prototyps for MsmtInfoReq, MsmtStopReq

Signed-off-by: monfron <simon.fronhoefer@gmail.com>

monfron committed:
Added: finalized udp-throughput (sendUdpMsmtInfoReq, sendUdpmsmtStopReq)

Signed-off-by: monfron <simon.fronhoefer@gmail.com>

monfron committed:
Implemented: WIP udp-measurement (almost finished c.f. notes)
- Somehow the close command is not received by the worker

Signed-off-by: monfron <simon.fronhoefer@gmail.com>

monfron committed:
Fixed: Usage of a single host-uuid for all measurement-modules

Signed-off-by: monfron <simon.fronhoefer@gmail.com>

monfron committed:
WIP: New udpServerWorker structure (dont block until read is running)
- this is done asynchronously
- BUT: now we run into the problem that we receive the conn.close cmd,
tear down the conn, but go routine asychronously still reads => use of closed network
(error)

- of course there are some "solutions" to that like:
	- break down the cause of the error and ignore "use of closed network" errors
	- or receive within the goroutine the termination cmd and close it there
- i dont like non of that

Signed-off-by: monfron <simon.fronhoefer@gmail.com>

monfron committed:
Added: Finer granularity for error handling at udp-throughput server

Signed-off-by: monfron <simon.fronhoefer@gmail.com>

monfron committed:
Fixed and tested: tcpServerWorker fixed so default case doesnt blocked too long

Signed-off-by: monfron <simon.fronhoefer@gmail.com>

hgn commented on mapago-client.go:
```diff
@@ -9,6 +9,7 @@ import "time"
 import "strconv"
 import "github.com/monfron/mapago/control-plane/ctrl/client-protocols"
 import "github.com/monfron/mapago/measurement-plane/tcp-throughput"
+import "github.com/monfron/mapago/measurement-plane/udp-throughput"
```
egal was ist monfron muss raus und durch die offizielle url ersetzt werden.

hgn commented on measurement-plane/tcp-throughput/server.go:
```diff
@@ -141,8 +143,46 @@ func (tcpMsmt *TcpMsmtObj) tcpServerWorker(closeCh <-chan interface{}, goHeartbe
+				os.Exit(1)
+			}
+
+			readCh <- bytes
```
Das ist ein Problem! Mit jeden Read() werden die bytes via channel an einen anderen thread (coroutine) übertragen, das ist viel zu starke last. NOGO

monfron replied:
hm die version mit dem default statement (polling channel) funktioniert aber ja auch ned (s. https://github.com/protocollabs/mapago/commit/80c4015b05098aed6f89fc4cb31926fd51e1e407#diff-8a1a07a95837840f189fd9f3ee65794eR170 )

monfron replied:
lass uns des mal zusammen besprechen...nicht das ich mir was überlege und dann findest du das shitty und die ganze arbeit war umsonst.

hgn commented on measurement-plane/udp-throughput/server.go:
```diff
@@ -166,8 +157,18 @@ func (udpMsmt *UdpThroughputMsmt) udpServerWorker(closeCh <-chan interface{}, go
 				cltAddrExists = true
 			}
 
+			readCh <- bytes
```
same same

monfron committed:
Fixed: Import path

Signed-off-by: monfron <simon.fronhoefer@gmail.com>

monfron committed:
Fixed + simplified: udp server throughput

- Added:
	- storage to save socks per stream
	- stream now directly writes to RESULT map
	- funcs to write and read to map which saves socks
- Removed:
	- channel for sending within goroutine
Signed-off-by: monfron <simon.fronhoefer@gmail.com>
